.github/workflows/build_and_deploy_docs.yaml

```diff
@@ -0,0 +1,94 @@
+name: Build and deploy docs to Github Pages
+
+on:
+  push:
+    branches:
+      - dev
+
+jobs:
+  build-saturn:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - name: Log details
+        run: |
+          BRANCH=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}
+          echo "Triggered on branch: $BRANCH"
+
+      - name: Set up Ruby (required for gem installation)
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '2.7'
+
+      - name: Install Asciidoctor to build docs
+        run: |
+          gem install asciidoctor
+          gem install asciidoctor-pdf
+          gem install rouge
+
+      - name: Build Docs - Collect needed files
+        run: |
+          set -e
+          PROJECT_FILES=(
+            "projects/saturn/src/main/resources/log4j2.properties"
+            "projects/saturn/src/main/resources/system-vocabulary.ttl"
+            "projects/saturn/taxonomies.ttl"
+            "projects/saturn/views.yaml"
+            "projects/saturn/vocabulary.ttl"
+          )
+          BUILD_DIR=./docs/build
+          version=$(cat VERSION)
+          
+          mkdir -p $BUILD_DIR/docs
+          cp ./README.adoc $BUILD_DIR
+          sed -i -e "s/VERSION/${version}/" $BUILD_DIR/README.adoc
+          cp -r ./docs/images $BUILD_DIR/docs/
+          for f in ${PROJECT_FILES[*]}; do
+          mkdir -p "$BUILD_DIR/$(dirname "$f")"
```

Comment:
I would put indent in this and next line... Not sure if there are some errors or warnings now without it, but for sure it would it better readable anyway

Reply:
it's been lost when copied this piece of code from local script which used for testing, added indentation and a few comment to the script